src/gso.ts

```diff
@@ -193,6 +194,74 @@ export class GSO {
     );
   }
 
+  /**
+   * Create an instruction for name tokens
+   * TODO: Once returning multiple insructions, merge this with config
+   */
+  public async createNameTokensInstruction(
+    projectName: string,
+    strikeAtomsPerLot: number,
+    authority: PublicKey,
+    baseMint: PublicKey,
+    quoteMint: PublicKey,
+    baseAccount: PublicKey,
+    quoteAccount: PublicKey,
+  ): Promise<web3.TransactionInstruction> {
+    const gsoState = await this.state(projectName);
+
+    const [soAuthority, _soAuthorityBump] = await web3.PublicKey.findProgramAddress(
+      [
+        Buffer.from(utils.bytes.utf8.encode('gso')),
+        gsoState.toBuffer(),
+      ],
+      this.program.programId,
+    );
+    const so = new StakingOptions(this.connection.rpcEndpoint);
+
+    const soState = await so.state(`GSO${projectName}`, baseMint);
+    const soOptionMint = await so.soMint(strikeAtomsPerLot, `GSO${projectName}`, baseMint);
+    const xBaseMint = await this.xBaseMint(gsoState);
+
+    const [optionMetadata, _optionMintMetadataBump] = (
+      await web3.PublicKey.findProgramAddress(
+        [
+          Buffer.from(utils.bytes.utf8.encode('metadata')),
+          metaplexId.toBuffer(),
+          soOptionMint.toBuffer(),
+        ],
+        metaplexId,
+      ));
+
+    const [xBaseMetadata, _xBaseMintMetadataAccountBump] = (
+      await web3.PublicKey.findProgramAddress(
+        [
+          Buffer.from(utils.bytes.utf8.encode('metadata')),
+          metaplexId.toBuffer(),
+          xBaseMint.toBuffer(),
+        ],
+        metaplexId,
+      ));
+
+    return this.program.instruction.nameTokens(
```

Comment:
Staking-options sdk complaining were missing an account. Double check everything is included for gso?

Reply:
It is included. Push hadnt finished yet.